### PR TITLE
feat: handle reCAPTCHA on Subscribe form

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds reCAPTCHA to Subscribe form submissions as implemented in https://github.com/Automattic/newspack-plugin/pull/1910, if available.

Also fixes a bug from #932 where the honeypot field is prefilled with an email address if one is known, which causes the form to always trigger the honeypot trap.

### How to test the changes in this Pull Request:

See testing instructions in https://github.com/Automattic/newspack-plugin/pull/1910.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/blocks/subscribe/index.php [hunk 1]
```diff
@@ -241,6 +248,15 @@ function process_form() {
 		return send_form_response( [ 'email' => \sanitize_email( $_REQUEST['email'] ) ] );
 	}
 
+	// reCAPTCHA test.
+	if ( ! empty( $_REQUEST['captcha_token'] ) && method_exists( '\Newspack\Reader_Activation', 'verify_captcha' ) ) {
```
Comment (miguelpeixe) [inline src/blocks/subscribe/index.php:252]: This check allows for the captcha test to be skipped if not sent. It must fail if it's enabled and the request doesn't provide a token.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
As mentioned in https://github.com/Automattic/newspack-plugin/pull/1910, I'm getting a javascript error on the block `view.js`:

```
Uncaught ReferenceError: regeneratorRuntime is not defined
```

Probably related to webpack being unable to properly compile the declared dependencies?

File: src/blocks/subscribe/index.php (hunk 1 above)
Comment (dkoo) [reply src/blocks/subscribe/index.php]: Updated in 37f658

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
I'm still having issues with the compiled `view.js`. It looks related to the polyfill and the usage of `async`:

```
Uncaught ReferenceError: regeneratorRuntime is not defined
```

I'm not sure how to resolve it dependency-wise. I got it working locally by refactoring to use `Promise` only and not `async/await`.

**Comment by dkoo** [comment]
[03b142f](https://github.com/Automattic/newspack-newsletters/pull/935/commits/03b142f4fd97e0937e546bfbedc3fbc3f623f9d0) replaces `async`/`await` with promises only.

File: src/blocks/subscribe/view.js [hunk 2]
```diff
@@ -13,37 +15,86 @@ import './style.scss';
...
+						return res( '' );
+					}
+
+					const captchaSiteKey = reCaptchaScript.getAttribute( 'src' ).split( '?render=' ).pop();
+					if ( ! grecaptcha?.ready || ! captchaSiteKey ) {
+						rej( __( 'Error loading the reCaptcha library.', 'newspack-newsletters' ) );
+					}
+
+					grecaptcha.ready( async () => {
+						const token = await grecaptcha
```
Comment (miguelpeixe) [inline src/blocks/subscribe/view.js:36]: There's still `async/await` here

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: src/blocks/subscribe/view.js (hunk 2 above)
Comment (dkoo) [reply src/blocks/subscribe/view.js]: Bah, sorry for the back and forth. b75f03a should fix it once and for all.

**Review by miguelpeixe — APPROVED** [review]
🙌

**Pull request merged**